lib/rubocop/cop/style/fetch_env_var.rb

```diff
@@ -97,11 +97,47 @@ def allowed_var?(node)
 
         def used_as_flag?(node)
           return false if node.root?
+          return true if used_if_condition_in_body(node)
 
+          node.parent.send_type? && (node.parent.prefix_bang? || node.parent.comparison_method?)
+        end
+
+        def used_if_condition_in_body(node)
           if_node = node.ancestors.find(&:if_type?)
-          return true if if_node&.condition == node
+          used_if_condition?(node, if_node&.condition) ||
+            used_with_other_condition?(node, if_node&.condition) ||
+            used_with_comparison_method?(node, if_node&.condition) ||
+            used_with_predicate_method?(node, if_node&.condition)
+        end
 
-          node.parent.send_type? && (node.parent.prefix_bang? || node.parent.comparison_method?)
+        def used_if_condition?(node, condition)
+          return false unless condition&.send_type?
+
+          condition&.child_nodes == node.child_nodes
+        end
+
+        def used_with_other_condition?(node, condition)
+          if condition&.or_type? || condition&.and_type?
+            condition&.child_nodes&.any?(node)
+          else
+            false
+          end
+        end
+
+        def used_with_comparison_method?(node, condition)
+          if condition&.send_type?
+            condition&.comparison_method? && condition&.child_nodes&.any?(node)
+          else
+            false
+          end
+        end
+
+        def used_with_predicate_method?(node, condition)
+          if condition&.send_type? && condition&.predicate_method?
+            condition&.child_nodes&.any?(node)
+          else
+            false
+          end
```

Comment:
I'm worried about the amount of save navigation and duplication of code. How about the following updates?

```diff
diff --git a/lib/rubocop/cop/style/fetch_env_var.rb b/lib/rubocop/cop/style/fetch_env_var.rb
index 69d24b626..c997b67cb 100644
--- a/lib/rubocop/cop/style/fetch_env_var.rb
+++ b/lib/rubocop/cop/style/fetch_env_var.rb
@@ -104,40 +104,21 @@ module RuboCop

         def used_if_condition_in_body(node)
           if_node = node.ancestors.find(&:if_type?)
-          used_if_condition?(node, if_node&.condition) ||
-            used_with_other_condition?(node, if_node&.condition) ||
-            used_with_comparison_method?(node, if_node&.condition) ||
-            used_with_predicate_method?(node, if_node&.condition)
-        end

-        def used_if_condition?(node, condition)
-          return false unless condition&.send_type?
+          return false unless (condition = if_node&.condition)
+          return true if condition.send_type? && (condition.child_nodes == node.child_nodes)

-          condition&.child_nodes == node.child_nodes
+          used_in_condition?(node, condition)
         end

-        def used_with_other_condition?(node, condition)
-          if condition&.or_type? || condition&.and_type?
-            condition&.child_nodes&.any?(node)
-          else
-            false
+        def used_in_condition?(node, condition)
+          if condition.send_type?
+            return false if !condition.comparison_method? && !condition.predicate_method?
+          elsif !condition.or_type? && !condition.and_type?
+            return false
           end
-        end

-        def used_with_comparison_method?(node, condition)
-          if condition&.send_type?
-            condition&.comparison_method? && condition&.child_nodes&.any?(node)
-          else
-            false
-          end
-        end
-
-        def used_with_predicate_method?(node, condition)
-          if condition&.send_type? && condition&.predicate_method?
-            condition&.child_nodes&.any?(node)
-          else
-            false
-          end
+          condition.child_nodes&.any?(node)
         end

         def offensive?(node)
```

Reply:
The updated code is below.

```ruby
         def used_if_condition_in_body(node)
           if_node = node.ancestors.find(&:if_type?)

           return false unless (condition = if_node&.condition)
           return true if condition.send_type? && (condition.child_nodes == node.child_nodes)

           used_in_condition?(node, condition)
         end

         def used_in_condition?(node, condition)
           if condition.send_type?
             return false if !condition.comparison_method? && !condition.predicate_method?
           elsif !condition.or_type? && !condition.and_type?
             return false
           end

           condition.child_nodes&.any?(node)
         end
```